config/src/comments.rs

```diff
@@ -274,12 +274,18 @@ fn comments() -> HashMap<String, String> {
 #how long a banned peer should stay banned
 #ban_window = 10800
 
-#maximum number of peers
-#peer_max_count = 125
+#maximum number of inbound peer connections
+#peer_max_inbound_count = 128
 
-#preferred minimum number of peers (we'll actively keep trying to add peers
-#until we get to at least this number
-#peer_min_preferred_count = 8
+#maximum number of outbound peer connections
+#peer_max_outbound_count = 8
```

Comment:
What's the point of this? If this is greater than peer_min_preferred_outbound_count, do we continue to connect to more outbound peers?

Reply:
`peer_max_outbound_count` lets us put a configurable limit on how many outgoing connections with peers we can have and by extension all connections. We continue to try to connect to more outbound peers until `peer_min_preferred_outbound_count` is reached (a "healthy peers mix"). By default these values are the same (we try to maintain connections on all available outgoing slots)

Reply:
But let's say you have `peer_min_preferred_outbound_count` of 8, and `peer_max_outbound_count` of 16, how would you ever end up with more than 8 connections?

Reply:
We may not after future refactoring to better clean up distinguishing incoming and outgoing connections. For now it happens when we broadcast compact blocks for example. There are probably other cases as well where outgoing connections are opened for one reason or another, exceeding the `peer_max_outbound_count` at least until `clean_peers()` is run to clean up any potential excess connections. In this case if `peer_max_outbound_count` was set to 16, I think we would hang on to those connections instead of dropping them when cleaning up peers.

Reply:
😕 You're saying we open outbound connections when broadcasting compact blocks? Where in the code does that take place, and what for?

Reply:
There are a few places in the code that will create temporary outgoing connections (I haven't verified it in broadcasting compact blocks in practice, but I think that is one area). You can set some test limits and watch logs to see the outgoing connections be temporarily exceeded. The hope of starting this process of distinguishing incoming and outgoing connections is to identify those places where we don't have full control over incoming/outgoing connections.